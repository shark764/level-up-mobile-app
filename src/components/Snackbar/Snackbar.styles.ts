import tailwind from '@utils/tailwind';
import { StyleSheet } from 'react-native';

const styles = {
  common: StyleSheet.create({
    common: {
      position: 'absolute',
      bottom: 0,
      borderRadius: 0,
      margin: 0,
      padding: 10
    }
  }),
  custom: {
    error: tailwind('bg-red-medium'),
    success: tailwind('bg-green-medium'),
    info: tailwind('bg-blue-medium'),
    warning: tailwind('bg-yellow-medium')
  }
};

export default styles;
