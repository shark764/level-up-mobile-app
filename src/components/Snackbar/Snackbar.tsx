import { SemanticColors } from '@app-types/appData';
import { Text } from '@components/Text';
import React, { ComponentProps } from 'react';
import { useWindowDimensions, Platform } from 'react-native';
import { Snackbar as PaperSnackbar } from 'react-native-paper';
import myStyles from './Snackbar.styles';

type SnackbarProps = ComponentProps<typeof PaperSnackbar> & {
  type: SemanticColors;
};
const Snackbar = ({ style, type, duration, ...props }: SnackbarProps) => {
  const { width } = useWindowDimensions();
  return (
    <PaperSnackbar
      duration={duration || 3000}
      style={[
        Platform.OS === 'ios' && { width },
        myStyles.common.common,
        myStyles.custom[type],
        style
      ]}
      {...props}>
      <Text type='body-semi'>{props.children}</Text>
    </PaperSnackbar>
  );
};

export default Snackbar;
