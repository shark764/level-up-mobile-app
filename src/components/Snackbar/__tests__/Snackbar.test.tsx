import React from 'react';
import Snackbar from '../Snackbar';
import renderer from 'react-test-renderer';

describe('<Snackbar />', () => {
  it('renders correctly', async () => {
    const result = renderer
      .create(
        <Snackbar type='success' visible={true} onDismiss={() => {}}>
          Hello world
        </Snackbar>
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
