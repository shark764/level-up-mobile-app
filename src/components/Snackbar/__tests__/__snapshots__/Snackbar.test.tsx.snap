// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLiveRegion="polite"
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(124, 183, 65, 1)",
        "borderRadius": 0,
        "bottom": 0,
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 0,
        "opacity": 0,
        "padding": 10,
        "position": "absolute",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 0.9,
          },
        ],
        "width": 750,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "marginLeft": 16,
              "marginVertical": 14,
            },
            Object {
              "color": "#ffffff",
              "marginRight": 16,
            },
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "MartianBThai-SemiBold",
              "fontSize": 14,
              "lineHeight": 23.8,
            },
            Object {},
          ]
        }
      >
        Hello world
      </Text>
    </Text>
  </View>
</RCTSafeAreaView>
`;
